scylla-cdc/src/stream_generations.rs

```diff
@@ -0,0 +1,540 @@
+        }
+
+        // Delete all leftovers from previous tests.
+        for table in vec![
+            TEST_STREAM_TABLE.to_string(),
+            TEST_GENERATION_TABLE.to_string(),
+        ] {
+            session
+                .query(format!("TRUNCATE {};", table), &[])
+                .await
+                .unwrap();
+        }
+    }
+
+    // Populate test tables with given data.
+    async fn populate_test_db(session: &Session) {
+        let stream_generation =
+            Timestamp(chrono::Duration::milliseconds(GENERATION_NEW_MILLISECONDS));
+
+        for generation in &[GENERATION_NEW_MILLISECONDS, GENERATION_OLD_MILLISECONDS] {
+            let mut query = Query::new(format!(
+                "INSERT INTO {} (key, time, expired) VALUES ('timestamps', ?, NULL);",
+                TEST_GENERATION_TABLE
+            ));
+            select_consistency(session, &mut query).await.unwrap();
+
+            session
+                .query(
+                    query,
+                    (Timestamp(chrono::Duration::milliseconds(*generation)),),
+                )
+                .await
+                .unwrap();
+        }
+
+        let mut query = Query::new(format!(
+            "INSERT INTO {}(time, range_end, streams) VALUES (?, -1, {{{}, {}}});",
+            TEST_STREAM_TABLE, TEST_STREAM_1, TEST_STREAM_2
+        ));
+        select_consistency(session, &mut query).await.unwrap();
+        session.query(query, (stream_generation,)).await.unwrap();
+    }
+
+    // Create setup for tests.
+    async fn setup() -> Result<GenerationFetcher, Box<dyn Error>> {
+        let uri = std::env::var("SCYLLA_URI").unwrap_or_else(|_| "127.0.0.1:9042".to_string());
+
+        let session = SessionBuilder::new().known_node(uri).build().await?;
+
+        create_test_db(&session).await;
+        populate_test_db(&session).await;
+
+        let generation_fetcher = GenerationFetcher::test_new(&Arc::new(session));
+
+        Ok(generation_fetcher)
+    }
+
+    #[tokio::test]
+    async fn test_fetch_all_generations() {
+        let fetcher = setup().await.unwrap();
+
+        let correct_gen = vec![
+            Generation(chrono::Duration::milliseconds(GENERATION_NEW_MILLISECONDS)),
+            Generation(chrono::Duration::milliseconds(GENERATION_OLD_MILLISECONDS)),
+        ];
+
+        let gen = fetcher.fetch_all_generations().await.unwrap();
+
+        assert_eq!(gen, correct_gen);
+    }
+
+    fn construct_timeuuid(milliseconds: i64) -> Uuid {
+        const CONTEXT_COUNT: u16 = 2137; // Just a random number.
+        let context = Context::new(CONTEXT_COUNT);
+
+        let time = std::time::Duration::from_millis(milliseconds as u64);
+        let timestamp = Timestamp::from_unix(context, time.as_secs(), time.subsec_nanos());
+
+        let u = Uuid::new_v1(timestamp, &[1, 2, 3, 4, 5, 6])
+            .expect("Couldn't create a valid uuid value.");
+        println!("{}", u);
+        u
```

Comment:
What is it?

Reply:
The last expression in a function serves as its return value - you do not need to write `return u;`. Here `u` is assigned to a variable in order to be printed out, then it is returned.